disnake/activity.py

```diff
@@ -347,6 +367,12 @@ def __init__(
             else try_enum(ActivityType, activity_type)
         )
 
+        self.status_display_type: Optional[StatusDisplayType] = (
+            status_display_type
+            if isinstance(status_display_type, StatusDisplayType)
+            else try_enum(StatusDisplayType, status_display_type)
```

Comment:
```suggestion
            try_enum(StatusDisplayType, status_display_type)
            if isinstance(status_display_type, int)
            else status_display_type
```
This would turn `status_display_type = None` into a `StatusDisplayType` member otherwise